src/styles/common/_grid.scss

```
@@ -0,0 +1,60 @@
+@import '../settings';
+@import '../colours';
+
+$gap: 1rem;
+$max-width: 100ch;
+$min-side-margin: 1rem;
+$grid-gap: 1em;
+$separator-width: 2px;
+$separator-offset: 5px;
+$n-columns: 12;
+
+/* uniprot-website grid system */
+.uniprot-grid {
+  display: grid;
+  grid-template-columns:
+        // left-margin
+
+    minmax($min-side-margin, calc((100% - #{$max-width}) / 2))
+    // main grid columns
+    repeat(12, 1fr)
+    // right-margin
+    minmax($min-side-margin, calc((100% - #{$max-width}) / 2));
+
+  & > .uniprot-grid-cell:first-child {
```

How about something like this to avoid having this rule taking over in the case we explicitely set on of the offset classes?
```suggestion
  & > .uniprot-grid-cell:not([class^="uniprot-grid-cell--offset-"]):first-child {
```

---

This is the type of CSS nightmare I have always been afraid of! Let me try and thanks for the suggestion!

---

I'd say you might be able to remove the need for the `uniprot-grid-cell` class name altogether then, as no offset class would default to this first-child rule, and adding a offset class would be required for any complex grid system with 2 or more rows

---

This only works if the offset class -- the one that we are excluding, is the first in the `calssName` attribute. I'm trying to find a way now to search the whole `className` rather only checking the start.

---

Just drop the magic then and make the use of an offset class required 🤷🏽‍♂️